lighthouse-core/lib/cache-map.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const isEqual = require('lodash.isequal');
+
+/**
+ * @fileoverview This class is designed to allow maps with arbitrary equality functions.
+ * It is not meant to be performant and is well-suited to use cases where the number of entries is
+ * likely to be small (like computed artifacts).
+ */
+module.exports = class CacheMap {
```

Comment:
for the record, I'm still not a huge fan of this name, something like `ArbitraryEqualityMap` would make more sense 😛

Reply:
I'm fine with calling it ArbitraryEqualityMap.